contracts/src/common/DedicatedResolver.sol

```
@@ -0,0 +1,300 @@
+            if ((contentType & contentTypes) != 0) {
+                data = _abis[contentType];
+                if (data.length > 0) {
+                    return (contentType, data);
+                }
+            }
+        }
+        return (0, "");
+    }
+
+    /// @dev Sets the implementer for an interface.
+    /// @param interfaceId The interface ID.
+    /// @param implementer The implementer address.
+    function setInterface(bytes4 interfaceId, address implementer) public onlyOwner {
+        _setInterface(interfaceId, implementer);
+    }
+
+    function _setInterface(bytes4 interfaceId, address implementer) private {
+        _interfaces[interfaceId] = implementer;
+        emit InterfaceChanged(NODE_ANY, interfaceId, implementer);
+    }
+
+    /// @dev Gets the implementer for an interface.
+    /// @param interfaceId The interface ID.
+    /// @return implementer The implementer address.
+    function interfaceImplementer(bytes32, bytes4 interfaceId) public view returns (address implementer) {
+        implementer = _interfaces[interfaceId];
+        if (implementer == address(0) && ERC165Checker.supportsInterface(addr(NODE_ANY), interfaceId)) {
+            implementer = address(this);
+        }
+    }
+
+    /// @dev Set the primary name.
+    /// @param _name The primary name.
+    function setName(string calldata _name) external onlyOwner {
+        _primary = _name;
+        emit NameChanged(NODE_ANY, _name);
+    }
+
+    /// @dev Get the primary name.
+    /// @return name_ The primary name.
+    function name(bytes32) external view returns (string memory name_) {
+        name_ = _primary;
+    }
+
+    /// @inheritdoc IDedicatedResolver
+    /// @dev True if `wildcard` or this resolver is exact in the registry.
+    function supportsName(bytes memory _name) public view returns (bool) {
+        if (wildcard) return true;
+        address impl = findResolverImplementation();
+        if (impl == address(0)) return false;
```

I think this should throw an error? Otherwise it will just mysteriously return empty results for all records.

---

We discussed this in Slack and chose the design where a non-zero default is set by construction.  I also added a hard `error ResolverFinderRequired()` check.